### ISSUE TITLE
Elven buildings

### ISSUE THREAD

**Issue by rquinio** [ajjr1996/Witcher#22]
Elven rulers can never improve their holdings: it says to improve one thing, you need Keep II, to get Keep II you need castle walls, and castle walls is not a buildable option in the castle holdings of elven rulers.

**Comment by JonStryker** [comment]
The reason for this was that Elves supposedly never built walls and where thus overrun by humans. I frankly do not remember who implemented it that way. 
But it is reasonable enough to me that Elves would start to add walls to their settlement after hundreds of years of conflict with humans and monsters ;)

**Comment by tannerflick4** [comment]
Are these still needing fixed?

**Comment by rquinio** [comment]
Yes ! Are you thinking of duplicating/adapting the castle building chain for dwarves and elves ? I don't think there's a way to replace only wall buildings without replacing also the buildings which depend on walls.

**Comment by tannerflick4** [comment]
Yes, I think having a duplicated yet adapted building set for Elves and Dwarves. Same kind of buildings but different bonuses they give.

**Comment by rquinio** [comment]
Yep pretty much, there's a short description here: https://ck2.paradoxwikis.com/Witcher_Kings#Races

Some things I could gather, to give some context (probably not all applicable): 

Dwarves:
- "Dwarven fortifications are unparalleled engineering marvels. Thanks to a system of furnaces, boilers, and reinforced gutters, every passageway into the fortress can change into a deadly trap in the blink of an eye."
"Mahakam is famous for its workshops, forges, and ironworks, which are renowned for producing the best steel in the world and, as a result, the finest weapons"
- They had catacombs

Elves:
- "Elven-crafted constructions survived the upheavals of war and gardens humans once burned down have grown back thanks to the efforts of their original inhabitants, who have returned here after morethan two centuries of exile."
- There is mention of an old elven aqueduct in Oxenfurt
- Some baths in elven ruins - https://witcher.fandom.com/wiki/C%C3%A1elmewedd
- Some elven sculptors had workshops: "the famed Kilkerinn Palace, which, according to legend, once housed the renowned artist Giann'Lorenzo's workshop. His sculptures now command exorbitant prices at auction"
- They had necropolis (https://witcher.fandom.com/wiki/Fen_Carn)
